core/src/main/java/org/elasticsearch/bootstrap/JNACLibrary.java

```diff
@@ -40,6 +40,7 @@
     public static final int ENOMEM = 12;
     public static final int RLIMIT_MEMLOCK = Constants.MAC_OS_X ? 6 : 8;
     public static final int RLIMIT_AS = Constants.MAC_OS_X ? 5 : 9;
+    public static final int RLIMIT_FSIZE = Constants.MAC_OS_X ? 1 : 1;
```

Comment:
I prefer it this way because it's consistent with how the other `RLIMIT` constants are defined where there are differences between macOS and Linux.